src/Runner/TestSuiteSorter.php

```diff
@@ -79,6 +141,38 @@ private function randomize(array $tests): array
         return $tests;
     }
 
+    private function sortDefectsFirst(array $tests): array
+    {
+        \usort($tests, [self::class, 'cmpDefectPriorityAndTime']);
+
+        return $tests;
```

Comment:
this array was already modified by `usort` under the hood, then it's returned...
with that, it's possible that one will write code:
```php
$sortedArray = $this->sortDefectsFirst($array);
```
and think that `$sortedArray !== $array`

Reply:
While it is true that `usort` changes the `$tests` we still need to pass that information back to the caller. We could pass `&$tests` but then the method would have a different signature than the others for random, reverse and resolve.